File: JPA_ONE-TO-ONE/src/main/java/com/one/service/UserService.java
package com.one.service;

import com.one.domain.PostRepository;
import com.one.domain.User;
import com.one.domain.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/**
 * created by Gyunny 2022/02/19
 */
@Transactional(readOnly = true)
@RequiredArgsConstructor
@Service
public class UserService {

    private final UserRepository userRepository;
    private final PostRepository postRepository;

    public void test() {
        userRepository.findAll().forEach(user -> {
            System.out.println(user.getPosts());
        });
    }

}

File: Spring_Session/src/main/kotlin/com/session/service/UserService.kt
package com.session.service

import com.session.dto.SignInDTO
import com.session.dto.SignUpDTO
import com.session.entity.User
import com.session.repository.UserRepository
import org.modelmapper.ModelMapper
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpSession

/**
 * created by Gyunny 2021/10/18
 */
@Service
class UserService(
    private val userRepository: UserRepository,
    private val modelMapper: ModelMapper,
    private val httpSession: HttpSession,
    private val httpServletRequest: HttpServletRequest
) {

    fun signIn(signInDTO: SignInDTO) {
        val user = userRepository.findByEmail(signInDTO.email)
        httpSession.setAttribute("userId", user.id)
        println(httpServletRequest.session)
        println(httpSession.getAttribute("userId"))
    }

    @Transactional
    fun signup(signUpDTO: SignUpDTO) {
        userRepository.save(modelMapper.map(signUpDTO, User::class.java))
    }

}